slackbot/bot.py

```diff
@@ -23,13 +31,15 @@ def __init__(self):
             bot_emoji=settings.BOT_EMOJI if hasattr(settings, 'BOT_EMOJI') else None
         )
         self._plugins = PluginsManager()
-        self._dispatcher = MessageDispatcher(self._client, self._plugins)
+        self._dispatcher = Dispatcher(self._client, self._plugins)
 
     def run(self):
         self._plugins.init_plugins()
         self._dispatcher.start()
         self._client.rtm_connect()
         _thread.start_new_thread(self._keepactive, tuple())
+        if py_schedule:
+            _thread.start_new_thread(self._scheduler, tuple())
```

Comment:
Instead of use a single dedicated thread to do this, we should reuse the [threadpool](https://github.com/lins05/slackbot/blob/f470accd3441e13595c7115896a492b83fb1f7ed/slackbot/dispatcher.py#L20). Please check how can the `scheduler` library be integrated into it.